### DISCUSSION TITLE
iBicha/UnityYoutubePlayer/tree/legacy/youtube-explode (Bug: TransientFailureException: Video watch page is broken.)

### DISCUSSION THREAD

**Discussion by mhadji05** [Tyrrrz/YoutubeExplode#554] [Q&A]
I use this project: https://github.com/iBicha/UnityYoutubePlayer/tree/legacy/youtube-explode

Tyrrrz/YoutubeExplode is the main dependency of this. Something changed in YouTube side the last days and we get this bug.
Please let me know hot to fix it.  

TransientFailureException: Video watch page is broken.
In most cases, this error indicates that the problem is on YouTube's side and this is not a bug in the library.
To resolve this error, please wait some time and try again.
If this issue persists, please report it on the project's GitHub page.
YoutubeExplode.ReverseEngineering.Responses.WatchPage+<>c__DisplayClass11_0+<<GetAsync>b__0>d.MoveNext () (at <5d494b727db1483bbc119b783047bb1a>:0)
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () (at <567df3e0919241ba98db88bec4c6696f>:0)
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) (at <567df3e0919241ba98db88bec4c6696f>:0)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) (at <567df3e0919241ba98db88bec4c6696f>:0)
System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) (at <567df3e0919241ba98db88bec4c6696f>:0)
System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () (at <567df3e0919241ba98db88bec4c6696f>:0)
YoutubeExplode.Internal.Retry+<WrapAsync>d__1`1[T].MoveNext () (at <5d494b727db1483bbc119b783047bb1a>:0)
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () (at <567df3e0919241ba98db88bec4c6696f>:0)
YoutubeExplode.Internal.Retry+<WrapAsync>d__1`1[T].MoveNext () (at <5d494b727db1483bbc119b783047bb1a>:0)
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () (at <567df3e0919241ba98db88bec4c6696f>:0)
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) (at <567df3e0919241ba98db88bec4c6696f>:0)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) (at <567df3e0919241ba98db88bec4c6696f>:0)
System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) (at <567df3e0919241ba98db88bec4c6696f>:0)
System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () (at <567df3e0919241ba98db88bec4c6696f>:0)
YoutubeExplode.ReverseEngineering.Responses.WatchPage+<GetAsync>d__11.MoveNext () (at <5d494b727db1483bbc119b783047bb1a>:0)
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () (at <567df3e0919241ba98db88bec4c6696f>:0)
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) (at <567df3e0919241ba98db88bec4c6696f>:0)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) (at <567df3e0919241ba98db88bec4c6696f>:0)
System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) (at <567df3e0919241ba98db88bec4c6696f>:0)
System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () (at <567df3e0919241ba98db88bec4c6696f>:0)
YoutubeExplode.Videos.Streams.StreamClient+<GetStreamContextFromWatchPageAsync>d__4.MoveNext () (at <5d494b727db1483bbc119b783047bb1a>:0)
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () (at <567df3e0919241ba98db88bec4c6696f>:0)
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) (at <567df3e0919241ba98db88bec4c6696f>:0)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) (at <567df3e0919241ba98db88bec4c6696f>:0)
System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) (at <567df3e0919241ba98db88bec4c6696f>:0)
System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () (at <567df3e0919241ba98db88bec4c6696f>:0)
YoutubeExplode.Videos.Streams.StreamClient+<GetManifestAsync>d__6.MoveNext () (at <5d494b727db1483bbc119b783047bb1a>:0)
--- End of stack trace from previous location where exception was thrown ---
System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () (at <567df3e0919241ba98db88bec4c6696f>:0)
System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) (at <567df3e0919241ba98db88bec4c6696f>:0)
System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) (at <567df3e0919241ba98db88bec4c6696f>:0)
System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) (at <567df3e0919241ba98db88bec4c6696f>:0)
System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () (at <567df3e0919241ba98db88bec4c6696f>:0)
YoutubePlayer.YoutubePlayer+<PlayVideoAsync>d__9.MoveNext () (at Assets/YoutubePlayer/Scripts/YoutubePlayer.cs:63)
UnityEngine.Debug:LogException(Exception)
YoutubePlayer.<PlayVideoAsync>d__9:MoveNext() (at Assets/YoutubePlayer/Scripts/YoutubePlayer.cs:80)
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1:SetException(Exception)
YoutubeExplode.Videos.Streams.<GetManifestAsync>d__6:MoveNext()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1:SetException(Exception)
YoutubeExplode.Videos.Streams.<GetStreamContextFromWatchPageAsync>d__4:MoveNext()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1:SetException(Exception)
YoutubeExplode.ReverseEngineering.Responses.<GetAsync>d__11:MoveNext()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1:SetException(Exception)
YoutubeExplode.Internal.<WrapAsync>d__1`1:MoveNext()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1:SetException(Exception)
YoutubeExplode.ReverseEngineering.Responses.<<GetAsync>b__0>d:MoveNext()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1:SetResult(String)
YoutubeExplode.ReverseEngineering.<GetStringAsync>d__6:MoveNext()
UnityEngine.UnitySynchronizationContext:ExecuteTasks() (at C:/buildslave/unity/build/Runtime/Export/Scripting/UnitySynchronizationContext.cs:104)

**Comment by Tyrrrz** [comment]
That sounds like an outdated error message

**Reply by mhadji05** [reply]
I tried to update the YoutubeExplode package by importing the following folder into my project: https://github.com/Tyrrrz/YoutubeExplode/tree/master/YoutubeExplode

Unfortunately I get a lot of errors. 
How can I update it ?
Αm Ι doing something wrong ?

**Reply by Tyrrrz** [reply]
I don't know, sorry. All I can suggest is to use NuGet.

**Reply by mhadji05** [reply]
Please, can you give me instructions on how to install the nugget?

**Reply by Tyrrrz** [reply]
I'm not sure how you would that in Unity or if Unity even supports it.

**Reply by jakedowns** [reply]
for anyone who finds this in the future, it is possible to install using NuGet, and then take the dll files and copy them from /Packages directory to the /Assets/Plugins directory. But i'm getting a different error now :( https://github.com/Tyrrrz/YoutubeExplode/discussions/679

It's partially working, cause it's printing the correct video title / duration. But when I try to access the streamInfo.Url, i get a Json-related error
